2020/day3.clj

```diff
@@ -7,27 +7,43 @@
        (string/split-lines)))
 
 (defn next-location
-  [loc [right down]]
-  [(+ (first loc) right)
-   (+ (second loc) down)])
+  [[row col] [right down]]
+  (let [next-row (+ row right)
+        next-col (+ col down)]
+    [next-row next-col]))
+
+(defn next-location-step
+  [right down n]
+  (next-location [0 0] [(* right n) (* down n)]))
```

Comment:
The fact that this is broken out indicates the earlier function doesn't need to exist. Combine these. Also, the name doesn't reflect what it's doing.... `next` implies an iteration, and that's not what should be occurring.

Reply:
This is the right idea. But it's arguments are inconsistent compared to functions below. i.e., why are right and down passed as single args? Aren't these a tuple?